src/pyhf/pdf.py

```diff
@@ -222,12 +222,22 @@ def __init__(self, spec, **config_kwargs):
             spec, self.channel_nbins
         )
 
-        poiname = config_kwargs.get('poiname', 'mu')
+        poiname = config_kwargs.pop('poiname', 'mu')
+
+        # measurement_name is explicitly removed in cases where the model is created using poi_name, multiple
+        # measurements are there but the first one is taken or if measurement is defined as None in the argument
+        if 'measurement_name' in config_kwargs:
+            del config_kwargs['measurement_name']
```

Comment:
Is the only reason this is done to ensure that `config_kwargs` is empty after the `"modifier_settings"` are popped off? If so, then this could probably be moved to just above the check and its positioning alone would make the comment unnecessary.

Reply:
Yes, I believe we can move it up. On second thoughts, wouldn't it be nice to keep the comment for future as we are already using pop while `_get_measurement()` is called?

Reply:
Pinging @matthewfeickert, just to highlight the above query.

Reply:
keep the comment and change to `pop`